src/providers/ruby.rs

```diff
@@ -0,0 +1,84 @@
+use super::Provider;
+use crate::nixpacks::{
+    app::App,
+    environment::{Environment, EnvironmentVariables},
+    nix::Pkg,
+    phase::{BuildPhase, InstallPhase, SetupPhase, StartPhase},
+};
+use anyhow::Result;
+use regex::Regex;
+
+enum Framework {
+    Rails,
+    /// No framework could be found
+    Vanilla,
+}
+
+pub struct RubyProvider {}
+
+impl Provider for RubyProvider {
+    fn name(&self) -> &str {
+        "ruby"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(app.includes_file("Gemfile") || app.has_match("*.rb"))
+    }
+
+    fn setup(&self, app: &App, _env: &Environment) -> Result<Option<SetupPhase>> {
+        let framework = self.detect_framework(app);
+        let mut packages = vec![Pkg::new("ruby")];
+        let needs_java = app.find_match(&Regex::new(r"jruby")?, "Gemfile.lock")?;
+        if needs_java {
+            packages.push(Pkg::new("java"));
+        }
+        match framework {
+            Framework::Rails => {
+                packages.append(&mut vec![Pkg::new("postgresql"), Pkg::new("nodejs")]);
```

Comment:
I think Rails applications have a `package.json` so we could get the node version from that. Would be something like `NodeProvider::get_nix_node_pkg`

Reply:
From an example of a rails app I cannot see a package.json in root. Are you sure there is one?

Reply:
Can use this https://github.com/cloudfoundry/ruby-buildpack/tree/master/fixtures for testing ruby provider.

All `rails5*` directores have `package.json`

Reply:
It exists in this Rails blog example https://github.com/railwayapp/starters/blob/master/examples/rails-blog/package.json

Not sure how node deps would be specified without one. Maybe it is optional?

Reply:
I've added node version and packaging support. Most of this logic should probably be cleaned in the Node provider, I might do that later on.